common/crypto/openssl/ec.c

```diff
@@ -440,6 +633,89 @@ oe_result_t oe_ec_public_key_from_coordinates(
 
     return result;
 }
+#else
+oe_result_t oe_ec_public_key_from_coordinates(
+    oe_ec_public_key_t* public_key,
+    oe_ec_type_t ec_type,
+    const uint8_t* x_data,
+    size_t x_size,
+    const uint8_t* y_data,
+    size_t y_size)
+{
+    oe_result_t result = OE_UNEXPECTED;
+    oe_public_key_t* impl = (oe_public_key_t*)public_key;
+    int nid;
+    EVP_PKEY* pkey = NULL;
+    EVP_PKEY_CTX* ctx = NULL;
+    OSSL_PARAM_BLD* param_bld = NULL;
+    OSSL_PARAM* params = NULL;
+
+    if (public_key)
+        oe_secure_zero_fill(public_key, sizeof(oe_ec_public_key_t));
+
+#if !defined(OE_BUILD_ENCLAVE)
+    /* Initialize OpenSSL */
+    oe_crypto_initialize();
+#endif
+
+    /* Reject invalid parameter */
+    if (!public_key || !x_data || !x_size || x_size > OE_INT_MAX || !y_data ||
+        !y_size || y_size > OE_INT_MAX)
+        OE_RAISE(OE_INVALID_PARAMETER);
+
+    /* Get the NID for this curve type */
+    if ((nid = _get_nid(ec_type)) == NID_undef)
+        OE_RAISE(OE_FAILURE);
+
+    /* Create the public EC key */
+    if (!(param_bld = OSSL_PARAM_BLD_new()))
+        OE_RAISE(OE_CRYPTO_ERROR);
+
+    if (!OSSL_PARAM_BLD_push_utf8_string(
+            param_bld, "group", OBJ_nid2sn(nid), 0))
+        OE_RAISE(OE_CRYPTO_ERROR);
+    unsigned char* pub_data = OPENSSL_malloc(1 + x_size + y_size);
+    if (!pub_data)
+        OE_RAISE(OE_OUT_OF_MEMORY);
+
+    pub_data[0] = (uint8_t)POINT_CONVERSION_UNCOMPRESSED;
+    memcpy(&pub_data[1], &x_data[0], x_size);
+    memcpy(&pub_data[1 + x_size], &y_data[0], y_size);
+
+    if (!OSSL_PARAM_BLD_push_octet_string(
+            param_bld, "pub", pub_data, 1 + x_size + y_size))
+        OE_RAISE(OE_CRYPTO_ERROR);
+
+    params = OSSL_PARAM_BLD_to_param(param_bld);
+
+    ctx = EVP_PKEY_CTX_new_from_name(NULL, "EC", NULL);
+    if (ctx == NULL || params == NULL || EVP_PKEY_fromdata_init(ctx) <= 0 ||
+        EVP_PKEY_fromdata(ctx, &pkey, EVP_PKEY_PUBLIC_KEY, params) <= 0)
+        OE_RAISE(OE_CRYPTO_ERROR);
+
+    OPENSSL_free(pub_data);
```

Comment:
Move the declaration of `pub_data` to the beginning of the function, and move this call after `done`, similar to other variables that needed to be freed.